prow/plugins/lgtm/lgtm.go

```diff
@@ -263,49 +279,54 @@ func handle(wantLGTM bool, config *plugins.Configuration, ownersClient repoowner
 	// Only add the label if it doesn't have it, and vice versa.
 	labels, err := gc.GetIssueLabels(org, repoName, number)
 	if err != nil {
-		log.WithError(err).Errorf("Failed to get the labels on %s/%s#%d.", org, repoName, number)
+		log.WithError(err).Errorf("Failed to get issue labels.")
 	}
 	hasLGTM := github.HasLabel(LGTMLabel, labels)
 
 	// remove the label if necessary, we're done after this
+	opts := optionsForRepo(config, rc.repo.Owner.Login, rc.repo.Name)
 	if hasLGTM && !wantLGTM {
 		log.Info("Removing LGTM label.")
-		return gc.RemoveLabel(org, repoName, number, LGTMLabel)
-	}
-
-	// add the label if necessary, we're done after this
-	if !hasLGTM && wantLGTM {
-		log.Info("Adding LGTM label.")
-		if err := gc.AddLabel(org, repoName, number, LGTMLabel); err != nil {
+		if err := gc.RemoveLabel(org, repoName, number, LGTMLabel); err != nil {
 			return err
 		}
-		// Delete the LGTM removed notification after the LGTM label is added.
-		botname, err := gc.BotName()
-		if err != nil {
-			log.WithError(err).Errorf("Failed to get bot name.")
+		if opts.StoreTreeHash {
+			cp.PruneComments(func(comment github.IssueComment) bool {
+				return addLGTMLabelNotificationRe.MatchString(comment.Body)
+			})
 		}
-		comments, err := gc.ListIssueComments(org, repoName, number)
-		if err != nil {
-			log.WithError(err).Errorf("Failed to get the list of issue comments on %s/%s#%d.", org, repoName, number)
+	} else if !hasLGTM && wantLGTM {
+		log.Info("Adding LGTM label.")
+		if err := gc.AddLabel(org, repoName, number, LGTMLabel); err != nil {
+			return err
 		}
-		for _, comment := range comments {
-			if comment.User.Login == botname && comment.Body == removeLGTMLabelNoti {
-				if err := gc.DeleteComment(org, repoName, comment.ID); err != nil {
-					log.WithError(err).Errorf("Failed to delete comment from %s/%s#%d, ID:%d.", org, repoName, number, comment.ID)
+		if opts.StoreTreeHash {
+			pr, err := gc.GetPullRequest(org, repoName, number)
+			if err != nil {
+				log.WithError(err).Errorf("Failed to get pull request.")
+			}
+			if pr.MergeSHA != nil {
+				commit, err := gc.GetSingleCommit(org, repoName, *pr.MergeSHA)
+				if err != nil {
+					log.WithError(err).Errorf("Failed to get commit.")
+				}
+				treeHash := commit.Commit.Tree.SHA
+				log.Info("Adding comment to store tree-hash: " + treeHash)
+				if err := gc.CreateComment(org, repoName, number, fmt.Sprintf(addLGTMLabelNotification, treeHash)); err != nil {
```

Comment:
Would it reduce spam if we edited the comment vs pruning and creating every time?